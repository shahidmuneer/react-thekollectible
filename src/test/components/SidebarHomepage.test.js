import {render} from '../utils'
import SidebarHomepage from '../../components/sidebarHomepage'
import React from 'react'

const dataMocks = ['butt', 'test', 'dragons', 'training'];

const onTagClick = jest.fn();


describe('<SidebarHomepage /> component', () => {
  const renderComponent = () => render(<SidebarHomepage onTagClick={onTagClick} listTag={dataMocks} />)

    it('renders sidebar title', () => {
        const wrapper = renderComponent()
        expect(wrapper.getByText('Popular Tags')).toBeDefined()
      })

    it('renders all tags', () => {
      const wrapper = renderComponent()
      const items= wrapper.container.getElementsByClassName('tag-pill')
      expect(items.length).toEqual(4)
      expect(wrapper.asFragment()).toMatchSnapshot()
    })
})