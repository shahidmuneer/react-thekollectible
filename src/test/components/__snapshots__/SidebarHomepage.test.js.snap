// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarHomepage /> component renders all tags 1`] = `
<DocumentFragment>
  <div
    class="sidebar"
  >
    <h3>
      Popular Tags
    </h3>
    <div
      class="tag-list"
    >
      <span
        class="tag-pill tag-default"
        style="cursor: pointer;"
      >
        butt
      </span>
      <span
        class="tag-pill tag-default"
        style="cursor: pointer;"
      >
        test
      </span>
      <span
        class="tag-pill tag-default"
        style="cursor: pointer;"
      >
        dragons
      </span>
      <span
        class="tag-pill tag-default"
        style="cursor: pointer;"
      >
        training
      </span>
    </div>
  </div>
</DocumentFragment>
`;
